pkg/syfs/syfs.go

```diff
@@ -43,20 +43,26 @@ func ConfigDir() string {
 }
 
 func configDir() string {
-	user, err := user.Current()
-	if err != nil {
-		sylog.Warningf("Could not lookup the current user's information: %s", err)
+	configDir := os.Getenv("SINGULARITY_CONFIGDIR")
+	if configDir == "" {
+		sylog.Warningf("Environment variable SINGULARITY_CONFIGDIR is not set")
 
-		cwd, err := os.Getwd()
+		user, err := user.Current()
 		if err != nil {
-			sylog.Warningf("Could not get current working directory: %s", err)
-			return singularityDir
+			sylog.Warningf("Could not lookup the current user's information: %s", err)
+
+			cwd, err := os.Getwd()
+			if err != nil {
+				sylog.Warningf("Could not get current working directory: %s", err)
+				return singularityDir
+			}
+
+			return filepath.Join(cwd, singularityDir)
 		}
 
-		return filepath.Join(cwd, singularityDir)
+		return filepath.Join(user.HomeDir, singularityDir)
 	}
-
-	return filepath.Join(user.HomeDir, singularityDir)
+	return configDir
```

Comment:
This is a long way from where `configDir` is extracted from the environment variable. It would be better to handle the case that the environment variable is set at the top of the function, returning as soon as possible.

That change would make the flow through the function easier to read, and remove one level of nesting.

Reply:
I thought that, too, but I didn't know if you would prefer a flat flow. Will change that.